### ISSUE TITLE
Start menu entries not created on the Windows side [lifted WSL only]

### ISSUE THREAD

**Issue by worldofgeese** [microsoft/wslg#600]
# Environment

```none
Windows build number: Win32NT 10.0.22000.0 Microsoft Windows NT 10.0.22000.0
Your Distribution version: Ubuntu 20.04 OR Distrod Arch Linux
Your WSLg version: 0.50.2.0
```

# Steps to reproduce
Using a bog-standard Ubuntu 20.04 or a Distrod supported distro running under the Windows Subsystem for Linux Preview installed from the Microsoft Store, then installing emacs using the standard package managers results in no Start Menu entries for any GUI Linux app. I can confirm that `Terminal=False` and have provided evidence below. 

**WSL logs**: 

[versions.txt](https://github.com/microsoft/wslg/files/7649490/versions.txt)
[weston.log](https://github.com/microsoft/wslg/files/7649488/weston.log)

#  Expected behavior

Start menu entries appear for any GUI Linux app.

# Actual behavior

No start menu entries for any Linux GUI app
![image](https://user-images.githubusercontent.com/59834693/144601068-8e71e0eb-c05b-497a-85d3-22926ce7ef60.png)

A `ls` of `/usr/share/applications`
![image](https://user-images.githubusercontent.com/59834693/144601742-0ba356aa-75dc-41bc-b941-24f812b369db.png)

![image](https://user-images.githubusercontent.com/59834693/144602285-1b702256-69b7-43f1-b876-bd939b8f7e00.png)

**Comment by dingdang66686** [comment]
After some research, I found a way to solve this problem. Just copy C:\Program Files\WindowsApps\MicrosoftCorporationII.WindowsSubsystemforLinux_0.50.2.0_x64__8wekyb3d8bbwe\WSLDVCPlugin.dll to C:\Windows\System32\ and open WSLDVCPlugin/RegisterWSLPlugin.reg from this repository. Reboot windows to take effect.

**Comment by FunWuChou** [comment]
> After some research, I found a way to solve this problem. Just copy C:\Program Files\WindowsApps\MicrosoftCorporationII.WindowsSubsystemforLinux_0.50.2.0_x64__8wekyb3d8bbwe\WSLDVCPlugin.dll to C:\Windows\System32\ and open WSLDVCPlugin/RegisterWSLPlugin.reg from this repository. Reboot windows to take effect.

非常感谢，通过这个方法，问题完美解决，你为何如此优秀~~

**Comment by hideyukn88** [comment]
@worldofgeese, @dingdang66686 and @FunWuChou, thanks for reporting the issue. I think I understand the issue but to confirm, please send us output from below command from Windows host, thanks!

`C:\>reg query "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Terminal Server Client\Default\OptionalAddIns" /S`

**Comment by dingdang66686** [comment]
<img width="927" alt="image" src="https://user-images.githubusercontent.com/42102822/148033754-53d9bc35-c1ac-48b6-96b8-8a7203e5587c.png">

**Comment by hideyukn88** [comment]
@dingdang66686, thanks for info, yes this does match with my understanding. The problem is that this plugin registration is only done when WSL is installed from Settings app, we call this "inbox WSL", but the registration process is missing when it is installed from Store app, we call this is "lifted WSL", thus anyone who never installed inbox WSL, will encounter this problem.

Thank you very much for all who reported this issue and provide information to us, we will make fix to lifted WSL installer and will make sure it will be included in next update, thanks!

**Comment by hideyukn88** [comment]
Fixed in WSLg 1.0.33
